prov/efa/src/rxr/rxr_ep.c

```diff
@@ -2641,17 +2692,10 @@ int rxr_endpoint(struct fid_domain *domain, struct fi_info *info,
 				fi_strerror(-retv));
 	}
 err_close_core_cq:
-	retv = fi_close(&rxr_ep->rdm_cq->fid);
+	retv = -ibv_destroy_cq(ibv_cq_ex_to_cq(rxr_ep->ibv_cq_ex));
 	if (retv)
 		FI_WARN(&rxr_prov, FI_LOG_CQ, "Unable to close cq: %s\n",
 			fi_strerror(-retv));
-err_close_shm_ep:
```

Comment:
Why remove this part?

Reply:
It is now unused. It was only used on line 2601 of the old file and I removed that.

Reply:
But your `rxr_ep_set_ibv_cq_ex(rxr_ep, &cq_attr);` did not clean shm resource in error path? You actually change the behavior of the code?

Reply:
fixed!